laalaa/apps/advisers/management/commands/add_disc_and_edu_categories.py

```diff
@@ -0,0 +1,21 @@
+from django.core.management.base import BaseCommand
+from advisers.models import Category
+
+
+class Command(BaseCommand):
+    help = "Add new Discrimination and Education categories."
+
+    def handle(self, *args, **options):
+        self.reset_adviser_category_pkey()
+        Category.objects.get_or_create(code="DISC", defaults={"civil": True})
+        Category.objects.get_or_create(code="EDU", defaults={"civil": True})
+
+    def reset_adviser_category_pkey(self):
+        # This is required because the id field is specified in the fixtures and has become out of sync with next
+        # sequence value
+        # Although I still need to understand if the sequence is being maintained by the database how specifying pkey in
+        # fixtures causes sequence to become out of sync
+        sql = 'SELECT setval(pg_get_serial_sequence(\'advisers_category\',\'id\'), coalesce(max("id"), 1), max("id") IS NOT null) FROM "advisers_category";'
```

Comment:
We tested the category sequence's next value on the environments:

| environment | has a sequence problem |
| --- | --- |
| local development | :warning: yes |
| CI environment (same as local) | :warning: yes |
| staging | :white_check_mark: was fine |
| production | :white_check_mark: was fine |

This command will not run automatically; on local environments and CircleCI the seeded database sequence will still be on "next value = 1" but on staging/production, it is not needed.

In conclusion, this `setval` query only needs to be included to pass local & CircleCI tests. Adding code for tests only works but hides the problem.

I think in order to make the new category additions consistent in all environments, we will need to move it to a migration entry. Categorical data, which cannot be edited by users, can be managed by automated migrations because the outcome is expected to be the same in all environments.

Reply:
What do you think about creating a new migration (and removing this command) that:

1. Fixes the sequence ID by resetting it like the above snippet.
1. Adds the two new categories.

If we do this, the list and IDs will be consistent from development to production.

Reply:
>1.Fixes the sequence ID by resetting it like the above snippet.
👍 

>Adds the two new categories.
Are we not going to have the same issues we had with cla_backend and the deployment failing due to multiple data migrations runs

Reply:
Maybe, but as the command's code is idempotent now, we could reuse that for the migration too.

Reply:
🎉 Changed command to migration